source/plugins/ruby/in_kubestate_deployments.rb

```diff
@@ -2,230 +2,237 @@
+      @ClusterId = KubernetesApiClient.getClusterId
+      @ClusterName = KubernetesApiClient.getClusterName
+    end
+
+    config_param :run_interval, :time, :default => 60
+    config_param :tag, :string, :default => Constants::INSIGHTSMETRICS_FLUENT_TAG
+
+    def configure(conf)
+      super
+    end
+
+    def start
+      if @run_interval
+        if !ENV["DEPLOYMENTS_CHUNK_SIZE"].nil? && !ENV["DEPLOYMENTS_CHUNK_SIZE"].empty?
+          @DEPLOYMENTS_CHUNK_SIZE = ENV["DEPLOYMENTS_CHUNK_SIZE"]
         end
+        $log.info("in_kubestate_deployments::start : DEPLOYMENTS_CHUNK_SIZE  @ #{@DEPLOYMENTS_CHUNK_SIZE}")
+
+        @finished = false
+        @condition = ConditionVariable.new
+        @mutex = Mutex.new
+        @thread = Thread.new(&method(:run_periodic))
       end
-  
-      def shutdown
-        if @run_interval
-          @mutex.synchronize {
-            @finished = true
-            @condition.signal
-          }
-          @thread.join
-        end
+    end
+
+    def shutdown
+      if @run_interval
+        @mutex.synchronize {
+          @finished = true
+          @condition.signal
+        }
+        @thread.join
       end
-  
-      def enumerate
-        begin
-          deploymentList = nil
-          currentTime = Time.now
-          batchTime = currentTime.utc.iso8601
-          
-          #set the running total for this batch to 0
-          @deploymentsRunningTotal = 0
-  
-          # Initializing continuation token to nil
-          continuationToken = nil
-          $log.info("in_kubestate_deployments::enumerate : Getting deployments from Kube API @ #{Time.now.utc.iso8601}")
-          continuationToken, deploymentList = KubernetesApiClient.getResourcesAndContinuationToken("deployments?limit=#{@DEPLOYMENTS_CHUNK_SIZE}", api_group: @DEPLOYMENTS_API_GROUP)
-          $log.info("in_kubestate_deployments::enumerate : Done getting deployments from Kube API @ #{Time.now.utc.iso8601}")
+    end
+
+    def enumerate
+      begin
+        deploymentList = nil
+        currentTime = Time.now
+        batchTime = currentTime.utc.iso8601
+
+        #set the running total for this batch to 0
+        @deploymentsRunningTotal = 0
+
+        # Initializing continuation token to nil
+        continuationToken = nil
+        $log.info("in_kubestate_deployments::enumerate : Getting deployments from Kube API @ #{Time.now.utc.iso8601}")
+        continuationToken, deploymentList = KubernetesApiClient.getResourcesAndContinuationToken("deployments?limit=#{@DEPLOYMENTS_CHUNK_SIZE}", api_group: @DEPLOYMENTS_API_GROUP)
+        $log.info("in_kubestate_deployments::enumerate : Done getting deployments from Kube API @ #{Time.now.utc.iso8601}")
+        if (!deploymentList.nil? && !deploymentList.empty? && deploymentList.key?("items") && !deploymentList["items"].nil? && !deploymentList["items"].empty?)
+          # debug logs to track the payload size
+          deploymentsSizeInKB = (deploymentList.to_s.length) / 1024
```

Comment:
you are calculating string length, not storage size with sizeof. Is this how you estimated payload size ?

Reply:
AFIK, there is no sizeof operator in ruby. are you referring https://apidock.com/ruby/String/bytesize as sizeof ?  btw, this is just estimate for the debug and will remove these as not needed for prod.